zio-http/src/main/scala/zio/http/ClientConfig.scala

```diff
@@ -13,6 +13,7 @@ case class ClientConfig(
   channelType: ChannelType = ChannelType.AUTO,
   nThreads: Int = 0,
   useAggregator: Boolean = true,
+  requestDecompression: (Boolean, Boolean) = (false, false),
```

Comment:
It might be nice to a have a case class for this pair o' bools, just so it's clearer which is which.

Reply:
In reality it has 3 values not 4 `(false, true)` and `(false, false)` are the same. 

We have 4 options
1. `case class Decompression(enabled: Boolean, strict: Boolean)`
2. `case class Decompression(strict: Boolean)` and to use `requestDecompression: Option[Decompression]`
3. `Option[Boolean]` - `None` - no decompressions and `Some(true)` - strict and `Some(false)` non strict
4. 
```sealed trait Decompression
object Decompression {
  case object NoDecompression extends Decompression
  case object StrictDecompression extends Decompression
  case object NonStrictDecompression extends Decompression
}
```
@kitlangton Which one do you prefer? Or do you have another?

Reply:
Use a sealed trait extended by 3 case objects instead of a case class. You can add one boolean method to that trait if desired.

Reply:
I changed it also in `ServerConfig`, the Boolean tuple was from there.